### PR TITLE
V10

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

* [ ] Adding `ObjectId`
* [ ] Adding `Decimal128`
* [ ] Adding embedded objects
* [ ] Integration with NG Sync

## ☑️ ToDos

* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 🚦 Tests
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [ ] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,176 @@
...
+### Enhancements
+* Add support for the `Decimal128` data type. ([#2674](https://github.com/realm/realm-js/issues/2674))
+* Add support for the `ObjectId` data type. ([#2675](https://github.com/realm/realm-js/issues/2675))
+* Add support for embedded objects. ([#2676](https://github.com/realm/realm-js/issues/2676))
+
+### Fixed
+* None.
+
+### Compatibility
+* Realm Object Server: 3.23.1 or later.
```
Comment (bmunkholm) [inline CHANGELOG.md:21]: I highly doubt this ;-)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,176 @@
...
+* File format: generates Realms with format v10 (reads and upgrades file format v5 or later).
+
+### Internal
+* Updated Realm Core from ??? to Realm Core v10.0.0-alpha.2.
+* Updated Realm Sync from ??? to Realm Sync v10.0.0-alpha.4.
+* Updated Realm Object Store to commit f5003ce365757bb2106ba4c317d98fd83b7f6c4e.
+* Created a package named "realm-app-importer", to be used by integration tests (ideally by other SDKs too).
+
+
+x.x.x Release notes (yyyy-MM-dd)
```
Comment (bmunkholm) [inline CHANGELOG.md:32]: ?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,176 @@
...
+* Add support for the `ObjectId` data type. ([#2675](https://github.com/realm/realm-js/issues/2675))
+* Add support for embedded objects. ([#2676](https://github.com/realm/realm-js/issues/2676))
+
+### Fixed
+* None.
+
+### Compatibility
+* Realm Object Server: 3.23.1 or later.
+* APIs are backwards compatible with all previous release of Realm JavaScript in the 4.x.y series.
+* File format: generates Realms with format v10 (reads and upgrades file format v5 or later).
```
Comment (bmunkholm) [inline CHANGELOG.md:23]:
```suggestion
* File format: generates Realms with format v11 (reads and upgrades file format v5 or later).
```

Comment (bmunkholm) [reply CHANGELOG.md:23]: We should also update the template for the release note to reflect above.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,176 @@
 x.x.x Release notes (yyyy-MM-dd)
 =============================================================
+NOTE 1: Requires MacOS v10.14 (Mojave) to build.
```
Comment (bmunkholm) [inline CHANGELOG.md:3]:
```suggestion
NOTE: This version does not support Realm Object Server or cloud.realm.io. Use v6.0 if you need that.
NOTE: This version bumps the Realm file format to version 11. Files created with previous versions of Realm will be automatically upgraded. It is not possible to downgrade to earlier versions.
```

File: CHANGELOG.md (hunk 1 above)
Comment (bmunkholm) [inline CHANGELOG.md:21]:
```suggestion
* Requires MacOS v10.14 (Mojave) to build.
* Sync functionality requires MongoDB Realm ??
```

File: CHANGELOG.md (hunk 4 above)
Comment (kneth) [reply CHANGELOG.md:3]: We have a pending update coming in from `v10-latest`.

**Comment by blagoev** [comment]
@kneth I guess this should be closed

**Pull request merged**